flax/linen/module.py

```diff
@@ -241,7 +241,11 @@ def wrap_method_once(fun: Callable[..., Any]) -> Callable[..., Any]:
     return fun
 
   @functools.wraps(fun)
-  def wrapped_module_method(self, *args, **kwargs):
+  def wrapped_module_method(*args, **kwargs):
+    if args and isinstance(args[0], Module):
```

Comment:
Can you add a comment explaining what this check does? I can't quite figure it out immediately. Is this to distinguish class methods from instance methods?

Reply:
Done

Reply:
Wouldn't it be clearer to only wrap the instance methods in `_wrap_module_methods` instead of within `wrapped_module_method`? (since it's a bit misleading -- sometimes the function that's returned from `wrapped_module_method` won't be wrapped after this change)